src/Microsoft.AspNetCore.Razor.Language/DefaultRazorParserOptions.cs

```diff
@@ -18,12 +18,18 @@ public DefaultRazorParserOptions(DirectiveDescriptor[] directives, bool designTi
             Directives = directives;
             DesignTime = designTime;
             ParseLeadingDirectives = parseLeadingDirectives;
+            Version = version ?? RazorParserFeatureContext.LatestRazorVersion;
+            FeatureContext = RazorParserFeatureContext.Create(Version);
         }
 
         public override bool DesignTime { get; }
 
         public override IReadOnlyCollection<DirectiveDescriptor> Directives { get; }
 
         public override bool ParseLeadingDirectives { get; }
+
+        public override Version Version { get; }
+
+        internal override RazorParserFeatureContext FeatureContext { get; }
```

Comment:
I would drop the name `Context` from this. It's not really acting as a context parameter. 

So maybe: `internal override RazorParserFeatureFlags FeatureFlags { get; set; }`